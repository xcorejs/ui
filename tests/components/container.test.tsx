import 'jest-styled-components';

import { Box, Container } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('Container component', () => {
  const component = renderer.create(
    <Container>
      <Box width="100%">
        <Box background="crimson" color="white" p="10px" my="10px">
          Box 1
        </Box>
        <Box background="navy" color="white" p="10px" my="10px">
          Box 2
        </Box>
      </Box>
    </Container>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
