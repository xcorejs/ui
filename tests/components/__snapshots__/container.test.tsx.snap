// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component 1`] = `
.c1 {
  width: 100%;
}

.c2 {
  padding: 10px;
  margin-top: 10px;
  margin-bottom: 10px;
  background: crimson;
  color: white;
}

.c3 {
  padding: 10px;
  margin-top: 10px;
  margin-bottom: 10px;
  background: navy;
  color: white;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
}

@media screen and (min-width:30em) {
  .c0 {
    width: 76.8rem;
  }
}

@media screen and (min-width:48em) {
  .c0 {
    width: 102.4rem;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    width: 120rem;
  }
}

@media screen and (min-width:78em) {
  .c0 {
    width: 132rem;
  }
}

<div
  className="c0"
  width={
    Array [
      "100%",
      "76.8rem",
      "102.4rem",
      "120rem",
      "132rem",
    ]
  }
>
  <div
    className="c1"
    width="100%"
  >
    <div
      className="c2"
      color="white"
    >
      Box 1
    </div>
    <div
      className="c3"
      color="white"
    >
      Box 2
    </div>
  </div>
</div>
`;
